Scheduler/feature_scheduler/auxtel/2022/03/fbs_config_sp_good_conditions.py

```diff
@@ -0,0 +1,195 @@
+import os
+
+import numpy as np
+
+import astropy.units as u
+
+from pathlib import Path
+
+from astropy.coordinates import Angle
+from astropy.io import ascii as astropy_ascii
+
+from rubin_sim.scheduler import basis_functions
+
+from rubin_sim.scheduler.utils import empty_observation
+from rubin_sim.scheduler.schedulers import Core_scheduler
+from rubin_sim.scheduler.surveys import Greedy_survey, FieldSurvey
+
+
+def get_basis_functions_cwfs_survey(
+    nside,
+    note,
+    time_gap_min,
+    wind_speed_maximum,
+):
+    """Get the basis functions for the CWFS survey.
+
+    This is a background survey that will activate at specific points in time
+    to make sure the telescope optics is aligned.
+
+    Parameters
+    ----------
+    nside : `int`
+        The nside value for the healpix grid.
+    note : `str`
+        A identifier string to be attached to the survey observations.
+    time_gap_min : `float`
+        The gap between observations of this survey, in minutes.
+    wind_speed_maximu : `float`
+        Maximum wind speed tolerated for the observations of the survey,
+        in m/s.
+
+    Returns
+    -------
+    `list` of `basis_functions.Base_basis_function`
+    """
+    sun_alt_limit = -12.0
+    return [
+        basis_functions.Not_twilight_basis_function(sun_alt_limit=sun_alt_limit),
+        basis_functions.Slewtime_basis_function(nside=nside, filtername="g"),
+        basis_functions.Slewtime_basis_function(nside=nside, filtername="r"),
+        basis_functions.Slewtime_basis_function(nside=nside, filtername="i"),
+        basis_functions.Moon_avoidance_basis_function(nside=nside),
+        basis_functions.Zenith_shadow_mask_basis_function(
+            min_alt=28.0, max_alt=85.0, nside=nside
+        ),
+        basis_functions.VisitGap(note=note, gap_min=time_gap_min),
+    ]
+
+
+def get_basis_functions_sp_survey(
+    ra,
+    nside,
+    note,
+    ha_limits,
+    wind_speed_maximum,
+    gap_min,
+):
+    # This will fallback to using the night boundaries specified on the
+    # driver.
+    sun_alt_limit = -12.0
+
+    return [
+        basis_functions.Not_twilight_basis_function(sun_alt_limit=sun_alt_limit),
+        basis_functions.Hour_Angle_limit_basis_function(RA=ra, ha_limits=ha_limits),
+        basis_functions.M5_diff_basis_function(nside=nside),
+        basis_functions.Slewtime_basis_function(nside=nside),
+        basis_functions.Moon_avoidance_basis_function(nside=nside),
+        basis_functions.Zenith_shadow_mask_basis_function(
+            min_alt=28.0, max_alt=85.0, nside=nside
+        ),
+        basis_functions.VisitGap(note=note, gap_min=gap_min),
+        basis_functions.AvoidDirectWind(
+            wind_speed_maximum=wind_speed_maximum, nside=nside
+        ),
+    ]
+
+
+def get_scheduler():
```

Comment:
Any chance you can add a doc string for this function? Please check the other new python files as well. You (understandably) seem to have done a lot of copying and pasting hehe